### ISSUE TITLE
MDPortMap documentation unclear

### ISSUE THREAD

**Issue by akorn** [icing/mod_md#228]
I find the following explanation unclear:

"With MDPortMap you can clarify on which local port http and https request arrive - should your server set behind a port mapper, such as an internet modem or a firewall.

If you use - for the local port, it indicates that this protocol is not available from the internet. For example, your Apache might listen to port 80, but your firewall might block it. mod_md needs to know this because it means that Let's Encrypt cannot send http: requests to your server."

A somewhat common use case is to have Apache listen on some high port(s) (in order to avoid having to start it with elevated privileges), and use e.g. `iptables` to forward connections to port 80 and 443 to the high ports used by Apache (say, 4080 and 4443).

Neither the above documentation, nor the one at https://httpd.apache.org/docs/2.4/mod/mod_md.html#mdportmap really explains whether MDPortMap is needed in this scenario; from Apache's perspective, the requests will arrive on port 4080 and 4443, so it appears `MDPortMap http:4080 https:4443` would be needed; however, Let's Encrypt can reach the server on port 80 and 443, so intuitively, MDPortMap should not be needed.

I'm not asking what to do (I can find that out by trial and error, and will in fact do so today); I would like the documentation to be more explicit.

For example, something like the following would be easier for me to understand: "MDPortMap is necessary if the http and/or https ports defined in your Apache configuration (in `<VirtualHost *:portnumber>` lines) are different from the ports these services are actually visible on from the Internet (e.g. because you use some DNAT mechanism to forward incoming connections to Apache). Use `MDPortMap http:port1 https:port2` if, from the Internet, the http service of your Apache instance is reachable on `port1` and the https service is reachable on `port2`."

Also, it's unclear what to do if Apache is listening on several http and/or https ports. https://httpd.apache.org/docs/2.4/mod/mod_md.html#mdportmap says the directive can only occur in the server config, not in a virtualhost context; however, some virtualhosts may use different port numbers (either internally or externally) than others.

**Comment by icing** [comment]
Does this help? <https://github.com/icing/mod_md#ports-ports-ports>

**Comment by akorn** [comment]
Yes, in fact I just got there. This part makes it clear: "For example, the firewall might forward all incoming connections to port 80 to the port 8888 of your Apache. In this case, mod_md should look for port 8888 instead of 80." -- I think this, or something like this (maybe even the wording I suggested) should be in the description of `MDPortMap`.

But even the "Ports Ports Ports" section doesn't address the issue with different http/https ports per VirtualHost.
